src/libs/math/include/math3d/color.h

```diff
@@ -18,49 +18,14 @@
 class Color
 {
   public:
-    union {
```

Comment:
What was the problem here?

Reply:
Current version gives error:
```
src/libs/math/include/math3d/color.h:38:32: error: member ‘Vector Color::<unnamed union>::<unnamed struct>::<unnamed union>::<unnamed union>::<unnamed struct>::c’ with constructor not allowed in anonymous aggregate
   38 |                         Vector c;
      |                                ^
```

GCC allows only:
https://gcc.gnu.org/onlinedocs/gcc/Unnamed-Fields.html
And GCC's implementation of the feature does not allow the unnamed struct to have members that have non-trivial constructors.

So this works in GCC:
```
union {
    struct
    {
        // Red
        float r;
        // Green
        float g;
        // Blue
        float b;
        // Transparency
        float a;
    };

    // Vector4 representation
    Vector4 v4;
};
```

And this is not:
```
union {
    struct
    {
        // rgb in vector
        Vector color;
        // Transparency
        float a;
    };

    // Vector4 representation
    Vector4 v4;
};
```

Reply:
That doesn't sound to be the case. Both classes have trivial(empty) constructors precisely for that reason.
Can you make a godbolt snippet?

Reply:
@espkk here you are: https://godbolt.org/z/8468cnoPs

Reply:
Move `default`ing into the class, that should work. I remember doing that, probably I missed it for Vector4

Reply:
IIRC, it's also possible to make a constructor inside the union to make default initialization explicit. I wouldn't avoid something like that, but the approach with `default` constructors should be totally fine and so far I don't see any non-standard or UB there

Reply:
@espkk ok. How to fix same error with non default constructor?
https://godbolt.org/z/caMfoEjTT

Reply:
@espkk I'm not sure how to do that. I thied `Vector color = new Vector();` - but it's not working: https://godbolt.org/z/aP5fqcjKd
Moving constructors into the class is not helping too for other constructors: https://godbolt.org/z/jaE3fz3dv

Reply:
Oh, gcc is so "special"
That one should work, but I'm not sure about UB https://godbolt.org/z/KavPdbf3r

Reply:
You fixed problem, when called `struct Foo`. I can add names to structs and added it in some commits. Is it worth to add names in the Color class?

Reply:
https://godbolt.org/z/8dcWfGY9j - here is more complex example. If remove `gcc_will_fail_if_removed` - gcc will fail.

Reply:
@espkk Hi. Is there any news/problems to solve for this PR?

Reply:
I'm happy with these changes